clients/src/main/java/org/apache/kafka/common/utils/Checksums.java

```diff
@@ -40,11 +55,41 @@ public static void update(Checksum checksum, ByteBuffer buffer, int length) {
     public static void update(Checksum checksum, ByteBuffer buffer, int offset, int length) {
```

Comment:
A java doc here which clarifies that this method leaves the buffer as it was received would be nice addition here.

Reply:
I can add some doc, although in theory it's unrelated to the changes of this PR; but is a nice addition :+1: